lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -91,8 +91,11 @@ def find_and_replace_metadata(request, resource, file)
   # @param resource [Puppet::Resource] The resource to add the metadata to
   # @param metadata [Puppet::FileServing::Metadata] The metadata of the given fileserver based file
   def replace_metadata(request, resource, metadata)
-    [:mode, :owner, :group].each do |param|
-      resource[param] ||= metadata.send(param)
+    file = resource.to_ral
+    if file.parameter(:source_permissions).value == :use
```

Comment:
I didn't account for checksum, so that would explain why the metadata is requested in all cases.